pkg/efivarfs/varfs.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2014-2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+//
+// This code is a somewhat simplified and changed up version of
+// github.com/canonical/go-efilib credits go their authors.
+// It allows interaction with the efivarfs via u-root which means
+// both read and write support.
+
+package efivarfs
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"syscall"
+
+	"github.com/u-root/u-root/pkg/guid"
+	"golang.org/x/sys/unix"
+)
+
+func init() {
+	if !probeEfivarfs() {
```

Comment:
I changed it from calling return to calling a panic. I know libraries and packages should never call a method that stops execution but in that case without having an efivarfs to begin with the whole package becomes ineffective, so that's why that check was there in the first place.